cache.go

```diff
@@ -22,7 +22,7 @@ import (
 
 // SecretBackend represents an interface for storing secrets.
 type SecretBackend interface {
-	Secret(string) (secret *Secret, ok bool)
+	Secret(string) (secret *Secret, deleted, ok bool)
```

Comment:
What would `deleted && ok` mean? Makes more sense to return `(*Secret, error)`, and then match on the exact error.